prov/gni/include/gnix_nic.h

```diff
@@ -52,6 +52,7 @@ extern "C" {
 #define GNIX_DEF_MAX_NICS_PER_PTAG	4
 
 extern uint32_t gnix_max_nics_per_ptag;
+extern uint32_t gnix_num_corespec_cpus;
```

Comment:
I don't like this being a global, in part because it's not clear whether I should use this value or `_gnix_get_num_corespec_cpus`.  I think the value should be cached in a file-local (static) global, and have all external uses call `_gnix_get_num_corespec_cpus`.  You might need a function static variable to indicate whether the value has been computed.